src/webots/nodes/WbAccelerometer.cpp

```diff
@@ -153,19 +153,20 @@ void WbAccelerometer::computeValue() {
   dBodyID upperSolidBodyId = upperSolid()->bodyMerger();
   if (upperSolidBodyId) {
     dVector3 newVelocity;
-    const WbVector3 &t = translation();
-    dBodyGetRelPointVel(upperSolidBodyId, t.x(), t.y(), t.z(), newVelocity);
-    const double InvDt = 1000.0 / mSensor->elapsedTime();
+    const WbVector3 &t = position();
+    dBodyGetPointVel(upperSolidBodyId, t.x(), t.y(), t.z(), newVelocity);
+    const double inverseDt = 1000.0 / mSensor->elapsedTime();
 
     for (int i = 0; i < 3; ++i) {
-      acceleration[i] += (newVelocity[i] - mVelocity[i]) * InvDt;
+      acceleration[i] += (newVelocity[i] - mVelocity[i]) * inverseDt;
       mVelocity[i] = newVelocity[i];
     }
-  }
+  } else
+    warn(tr("Parent of Accelerometer node has no physics: measurements may be wrong."));
```

Comment:
This message is printed everytime `wb_accelerometer_get_values` is called.
This is quite annoying and useless in controllers where the function is called in the main loop (for example `e-puck_kinematic.wbt`).
Personally I think that the warning should be printed only once.
Even if the measurements are not precise a user could want to get them, otherwise we should forbid it instead of continuously printing warnings.

Reply:
In this case, the measurement may be correct or totally wrong. It is not a problem of precision. But I agree, the warning should be printed only once. I will fix it and return nan values this case to avoid that the user relies on possibly wrong values.